R/diagnose_design.R

```diff
@@ -94,18 +102,21 @@ diagnose_design <- function(...,
     simulations_df <- simulate_design(!!!dots, sims = sims)
     diagnosands <- setup_diagnosands(!!!dots, diagnosands = diagnosands)
   }
-
+  
   # figure out what to group by ---------------------------------------------
+  # Optionally modify the simulations dataframe to create new grouping variables
+  if(!is.null(make_groups)) simulations_df <- 
+      eval(parse(text = paste("fabricate(simulations_df, ", make_groups, ")")))
```

Comment:
Instead of base NSE using eval/parse - use rlang quosures and splice in make_groups.

Something like

    eval_tidy(quo(fabricate(simulations_df, !!make_groups)))